mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -0,0 +1,312 @@
+import logging
+import json
+import os
+from collections import defaultdict
+from time import time
+from typing import Dict, Tuple
+import tqdm
+
+from datasets import load_dataset, Value, Features
+
+from ..evaluation.evaluators import InstructionRetrievalEvaluator
+from ..evaluation.evaluators import utils
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+# Adapted from https://github.com/beir-cellar/beir/blob/f062f038c4bfd19a8ca942a9910b1e0d218759d4/beir/datasets/data_loader_hf.py#L10
+class HFDataLoader:    
```

Comment:
I had to modify this class because it now comes with a bunch of extra fields, such as the instructions, the changed_instructions, the top_ranked, etc.